### DISCUSSION TITLE
Run multiple times on multiple lines with one command

### DISCUSSION THREAD

**Discussion by samawelch** [chhoumann/quickadd#660] [Q&A]
Hi, thanks for all your work on the plugin. Is it possible to, with one command, run a macro on multiple lines, treating each one as a new instance?

So that 
`Function (
Item 1
Item 2
)`

would do the same thing as 
`Function (Item 1)
Function (Item 2)`

just by selecting both items and running the relevant (in this case template) macro.

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

There's no built-in "run once per selected line", but a small user script in a Macro choice does exactly what you described: split the selection into lines and run your choice once per line, each as its own instance.

```js
module.exports = async (params) => {
  const editor = params.app.workspace.activeEditor?.editor;
  if (!editor) return;
  const lines = editor
    .getSelection()
    .split("\n")
    .map((l) => l.trim())
    .filter(Boolean);
  for (const line of lines) {
    await params.quickAddApi.executeChoice("Your Choice Name", { value: line });
  }
};
```

Passing `{ value: line }` pre-fills the plain `{{VALUE}}` token in the target choice, so each line runs without any prompting. If your choice uses a named variable like `{{VALUE:item}}`, pass `{ item: line }` instead.

Setup: create a Macro choice, add this script as a step, bind the macro to a hotkey, select your lines, run.

`executeChoice` reference: https://quickadd.obsidian.guide/docs/QuickAddAPI
